src/checkout-buttons/strategies/apple-pay/apple-pay-button-strategy.ts

```diff
@@ -216,12 +217,25 @@ export default class ApplePayButtonStrategy implements CheckoutButtonStrategy {
         const { currency: { decimalPlaces } } = state.cart.getCartOrThrow();
         let checkout = state.checkout.getCheckoutOrThrow();
         const availableOptions = checkout.consignments[0].availableShippingOptions;
-        const shippingOptions = availableOptions?.map(option => ({
+        const selectedOption = availableOptions?.filter(option => option.id === checkout.consignments[0].selectedShippingOption?.id);
+        const unSelectedOptions = availableOptions?.filter(option => option.id !== checkout.consignments[0].selectedShippingOption?.id);
+        const firstShippingOption = selectedOption?.map(option => ({
             label: option.description,
-            amount: `${option.cost.toFixed(decimalPlaces)}`,
+            amount: option.cost.toFixed(decimalPlaces),
             detail: option.additionalDescription,
             identifier: option.id,
         }));
+        const nonSelectedShippingOptions = unSelectedOptions?.map(option => ({
```

Comment:
```suggestion
        const restShippingOptions = unSelectedOptions?.map(option => ({
```

Reply:
Hi @clopezh, thanks for your suggestion, for sure I will change it.